Wrappers/Python/cil/optimisation/algorithms/PDHG.py

```diff
@@ -166,11 +292,106 @@ def update(self):
 
         self.g.proximal(self.x_tmp, self.tau, out=self.x)
 
-        #update_previous_solution() called after update by base class
-        #i.e current solution is now in x_old, previous solution is now in x
+        # update_previous_solution() called after update by base class
+        #i.e current solution is now in x_old, previous solution is now in x        
+
+        # update the step sizes for special cases
+        self.update_step_sizes()
+
+
+    def set_step_sizes(self, sigma=None, tau=None):
+
+        """Default step sizes for the PDHG algorithm"""
         
+        # Compute operator norm
+        self.norm_op = self.operator.norm()
+
+        if tau is None and sigma is None:            
+            self._sigma = 1./self.norm_op
+            self._tau = 1./self.norm_op
+        elif tau is None:
+            self._tau = 1./(self.sigma*self.norm_op**2)
+        elif sigma is None:
+            self._sigma = 1./(self.tau*self.norm_op**2)
+        else:
+            self._sigma = sigma
+            self._tau = tau
+      
+    def update_step_sizes(self):
```

Comment:
This should be `_update_step_sizes()` if it's a method for internal use by the class